src/file-upload/file-item.tsx

```diff
@@ -0,0 +1,182 @@
+    ItemText,
+    LoadingActionContainer,
+    MobileErrorMessage,
+} from "./file-item.styles";
+import { FileUploadHelper } from "./helper";
+import { FileItemProps } from "./types";
+
+interface Props extends FileItemProps {
+    onDelete: () => void;
+    onEdit?: (() => void) | undefined;
+}
+
+export const FileItem = ({
+    id,
+    name,
+    size,
+    type,
+    description,
+    progress = 1,
+    errorMessage,
+    thumbnailImageDataUrl,
+    editableMode,
+    truncateText = true,
+    onDelete,
+    onEdit,
+}: Props) => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const [fileName, setFileName] = useState<string>();
+    const isLoading = progress < 1;
+
+    const nameSectionRef = useRef<HTMLDivElement>();
+    const { width: wrapperWidth, ref: wrapperRef } = useResizeDetector();
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        setFileName(getTruncatedText(name));
+    }, [wrapperWidth]);
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleDelete = () => {
+        onDelete();
+    };
+
+    const handleEdit = () => {
+        if (onEdit) onEdit();
+    };
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    const getTruncatedText = (value: string) => {
+        if (!truncateText) return value;
+
+        const widthOfElement =
+            nameSectionRef && nameSectionRef.current
+                ? nameSectionRef.current.getBoundingClientRect().width
+                : 0;
+
+        return StringHelper.truncateOneLine(
+            value,
+            widthOfElement,
+            widthOfElement / 2,
+            widthOfElement / 2 / 8, // Arbitrary
+            16 // Font size
+        );
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderActionButton = () => {
+        if (isLoading) {
+            return (
+                <LoadingActionContainer>
+                    <ProgressBar progress={progress} />
+                </LoadingActionContainer>
+            );
+        } else if (errorMessage) {
+            return (
+                <ItemActionContainer>
+                    <ErrorIconButton
+                        onClick={handleDelete}
+                        data-testid={`${id}-error-delete-button`}
+                        aria-label={`delete-${name}`}
```

Comment:
voiceover reads it as "delete dash image.png", best to keep as a separate word
```suggestion
                        aria-label={`delete ${name}`}
```